framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/model/ServerManagementModel.java

```diff
@@ -171,9 +185,67 @@ protected IJobManager createJobManager() {
 
 	@Override
 	public void dispose() {
+		shutdownAllServers();
 		if( this.jobManager != null ) {
 			this.jobManager.shutdown();
 		}
 	}
 
+	protected void shutdownAllServers() {
+		if( getNotStoppedServers().isEmpty())
+			return;
+		
+		System.out.println("Shutting down started servers");
+		shutdownServers(getStartedServers(), false);
+		System.out.println("Shutting down all servers");
+		shutdownServers(getNotStoppedServers(), true);
+	}
+	
+	protected void shutdownServers(List<IServer> list, boolean force) {
+		if( list.isEmpty()) 
+			return;
+		
+		ExecutorService threadExecutor = Executors.newFixedThreadPool(list.size());
+		CountDownLatch latch = new CountDownLatch(list.size());
+		Iterator<IServer> it = list.iterator();
+		while(it.hasNext()) {
+			final IServer next = it.next();
+			System.out.println("Shutting down " + next.getId());
+
+			threadExecutor.submit( new Runnable() {
```

Comment:
runnable could be expressed in a more concise way using a lambda. And then I'd also consider extracting the runnable/lambda to a method so that ``` #shutdownServers``` gets shorter and better readable.